crates/compiler/can/src/annotation.rs

```diff
@@ -1192,8 +1192,10 @@ fn can_extension_type(
             );
 
             if valid_extension_type(shallow_dealias_with_scope(scope, &ext_type)) {
-                if matches!(loc_ann.extract_spaces().item, TypeAnnotation::Wildcard)
-                    && matches!(ext_problem_kind, ExtensionTypeKind::TagUnion)
+                if matches!(
+                    loc_ann.extract_spaces(env.arena).item,
```

Comment:
This should probably be changed to use `loc_ann.without_spaces()`, since we don't care about the spaces part. Ditto for other cases where we discard the spaces. This way we can avoid unnecessary allocation.

I'm ok doing that in this PR or in a separate PR.